README.md

```diff
@@ -124,11 +124,65 @@ caching by way of expiry or ETags, I recommend using `faraday-http-cache` for
 your clients and setting `memoize` to `false`. Remote Record will eventually
 gain support for caching.
 
+### `remote_all`
+
+If you're able to fetch multiple records at once from the API, implement the
+`self.all` method on your remote record class. This should return an array of
+hashes that can be used to initialize a set of references.
+
+This can optionally take a block
+for authorization - note that it won't use the auth you've configured and that
+you'll always have to supply that inline. For example:
+
+```ruby
+module RemoteRecord
+  module GitHub
+    # :nodoc:
+    class User < RemoteRecord::Base
+      def get
+        client.user(remote_resource_id)
+      end
+
+      def self.all
+        Octokit::Client.new(access_token: yield).users
+      end
+
+      private
+
+      def client
+        Octokit::Client.new(access_token: authorization)
+      end
+    end
+  end
+end
+```
+
+Now you can call `remote_all` on remote reference classes that use
+`RemoteRecord::GitHub::User`, like this:
+
+```ruby
+GitHub::UserReference.remote_all { GITHUB_PERSONAL_ACCESS_TOKEN }
```

Comment:
I don't love the idea of adding a `remote_all` method and passing a block, it doesn't seem obvious enough.

Is there a way we could instead use active record for this kind of work:

```ruby
GitHub::UserReference.where([conditions]).fetch_remote_records
```

To achieve that kind of an API you'd have to extend other parts of active record but the idea is that you would fetch all of the references with an activerecord query and then when you actually execute the query you could somehow batch the API calls into a bulk API if one exists.

Reply:
I'm planning to do something like this after this PR — this is something we need internally. Figuring out how to do `all` takes care of a lot of the heavy lifting for that.

Right now it's possible to filter `remote_all` using `Enumerable#select` and similar methods, but things like `where().fetch_remote_records` will still be an important thing to define so that less is done in memory and more is entrusted to the API.

In reality we've got two angles to approach it from — selecting multiple IDs to request from the API using a bulk endpoint, and using search functions of an API to filter down the array of references before they can be constructed. The latter is what we need for TimeKit — we want to call `bookings?search=some.attribute:some_value;some_other.attribure:some_value` and have references returned.